utils/http/utils.go

```diff
@@ -54,3 +55,98 @@ func GetConsoleApiToken(console, port string) (string, error) {
 
 	return token.GetApiToken(), nil
 }
+
+type StreamingConfig struct {
+	Addition    bool
+	Deletion    bool
+	EntityTypes []string //Node, Container, ContainerImage, Pod, KubernetesCluster, CloudNode
+}
+
+func (cfg StreamingConfig) validateConfig() error {
+	if cfg.Addition == false && cfg.Deletion == false {
+		return fmt.Errorf("Invalid StreamingConfig, should have atleast one event set")
+	}
+
+	validEntityTypes := make(map[string]bool)
+	//We only support streaming for entities of type
+	//Host, Container and ContainerImage
+	validEntityTypes["Node"] = true
+	validEntityTypes["Container"] = true
+	validEntityTypes["ContainerImage"] = true
+
+	for _, entity := range cfg.EntityTypes {
+		if _, ok := validEntityTypes[entity]; !ok {
+			return fmt.Errorf("Invalid entity in request:" + entity)
+		}
+	}
+
+	return nil
+}
+
+func StartStreamingDelta(ctx context.Context, cfg StreamingConfig,
+	url, port, apiToken string,
+	callback func(bool, []openapi.ModelNodeIdentifier)) error {
+	//First validate the config
+	err := cfg.validateConfig()
+	if err != nil {
+		return err
+	}
+
+	apiClient := NewHttpsConsoleClient(url, port)
+	if apiClient == nil {
+		return fmt.Errorf("Failed to create console client")
+	}
+
+	err = apiClient.APITokenAuthenticate(apiToken)
+	if err != nil {
+		return err
+	}
+
+	//Client is authenticated and we have valid config
+	//Start the streaming loop
+	interval := int64(30) //In seconds
+	ticker := time.NewTicker(time.Duration(interval) * time.Second)
+	timestamp := time.Now().UnixMilli()
+
+	topologyAPI := apiClient.Client().TopologyAPI
+	req := topologyAPI.GetTopologyDelta(ctx)
+	deltaReq := openapi.ModelTopologyDeltaReq{}
+	deltaReq.SetAddition(cfg.Addition)
+	deltaReq.SetDeletion(cfg.Deletion)
+	deltaReq.SetEntityTypes(cfg.EntityTypes)
+
+streamloop:
+	for {
+		select {
+		case <-ctx.Done():
+			break streamloop
+		case <-ticker.C:
+			deltaReq.SetTimestamp(timestamp)
+			req = req.ModelTopologyDeltaReq(deltaReq)
+			deltaRes, rh, err := topologyAPI.GetTopologyDeltaExecute(req)
+			if err != nil {
+				//TODO:: Should we return on error?
```

Comment:
Two ways of dealing with errors:
- Either we ask for an error callback that we call every time an error occurs
- Or we return a `chan error` and send the error every time it occurs.

Since we started with callbacks API, first option is probably the best one

Reply:
Done.